alt_e2eshark/onnx_tests/models/model.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+import os
+import numpy
+import onnx
+import torch
+from onnxruntime.tools.onnx_model_utils import make_dim_param_fixed, fix_output_shapes
+from e2e_testing import azutils
+from e2e_testing.framework import OnnxModelInfo
+from e2e_testing.registry import register_test
+from e2e_testing.storage import TestTensors
+from .protected_list import protected_models, models_with_postprocessing
+
+class AzureDownloadableModel(OnnxModelInfo):
+    def __init__(self, name: str, onnx_model_path: str, cache_dir: str):
+        opset_version = 21
+        super().__init__(name, onnx_model_path, cache_dir, opset_version)
+
+    def construct_model(self):
+        # try to find a .onnx file in the test-run dir
+        # if that fails, check for zip file in cache
+        # if that fails, try to download and setup from azure, then search again for a .onnx file
+
+        # TODO: make the zip file structure more uniform so we don't need to search for extracted files
+        model_dir = self.model.rstrip("model.nx")
```

Comment:
Is this supposed to be model.onnx?

Reply:
It should work either way. E.g.
```python
s = "abbc"
print(s.rstrip("cb")) # prints "a"
print(s.rstrip("bbc")) # also prints "a"
```
I'm fine changing it to "model.onnx" for readability.